pkg/service/keystore/service.go

```diff
@@ -46,6 +46,22 @@ func (s Service) Config() config.KeyStoreServiceConfig {
 	return s.config
 }
 
+// NewKeyStoreFromKeystoreStorage uses  a keystore service directly from storage object
+func NewKeyStoreFromKeystoreStorage(config config.KeyStoreServiceConfig, keyStoreStorage *Storage) (*Service, error) {
```

Comment:
where would this key store be coming from in practice?

Reply:
@decentralgabe so it's a good question. I was writing some personal stuff for testing around this and I needed an entry point that did not generate a new service key and salt, which is why I needed a method to support this action. I'm happy to just support this entrypoint on my own fork of this, or if there's a better way to handle this I'm open to it.

Reply:
what about adding a parameter to the config to not generate a new key and instead look for an existing one?

Reply:
@decentralgabe I think that's separate concerns. How it's managed in the config (optional/default behavior) is certainly important, but in the end of the day you'd still need to provide the logic in the NewKeyStoreService function to ignore re-generating the service keys, and/or to load the existing one. You then need to address the behavior of how they are loaded internally, etc, which was out of scope for this PR.  Certainly a valid point, just not sure if that's addressed best somewhere else.

Reply:
I checked if `ssi-service` key was set, and if it was, loaded that, using the method above. There are other possible ways to handle this, which is why I didn't want to tackle the logic around configuration/loading existing keys in this PR.

Reply:
This parameters of this new constructor makes sense to me. I recommend only having a single constructor, instead of 2. It seems to me that it makes sense to update all callers to use the new constructor, and remove the old one.

Keeping a single constructor can then be named `NewKeyStoreService`.

Reply:
Yea..@andresuribe87 that works and I think is a reasonable way forward.